src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -476,6 +476,11 @@ public static bool IsPossibleArrayGenericInterface(this TypeSymbol type)
             return false;
         }
 
+        internal static bool IsValidScopedType(this TypeSymbol? type)
```

Comment:
>IsValidScopedType [](http://example.com/codeflow?start=29&length=17)

I think the name is somewhat confusing. Neither null, nor error types are valid types. I suspect this helper is used to suppress diagnostics somewhere, but it is placed in a place where it is considered a general purpose helper and has behavior that might not be suitable in a general case.